### DISCUSSION TITLE
Dynamic sitemap for next.js

### DISCUSSION THREAD

**Discussion by rbhatt1999** [vercel/community#3720] [Help]
### Question

I'm using next.js 13.3 where I created the sitemap.js in app directly and I used a async call to a function to get the url for posts( I've also used no cache). But the while deploying it, vercel is generating the sitemap only while building but not on the runtime. Because of this when I add a post it will not reflect in the sitemap please help to solve the issue.

**Comment by amyegan** [comment]
Hey @rbhatt1999. It sounds like you followed the steps from the Next.js docs to [generate a sitemap](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/sitemap#generate-a-sitemap). 

Can you share your public repo or a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)? That will give us a better understanding of the problem.

**Reply by rbhatt1999** [reply]
here is the repo - 
https://github.com/rohitpeepoye/peepoyenetwork

@amyegan , I've used no cached for 2 different types of post. It is still creating sitemap as a static page at build time, not a server render at runtime
![image](https://github.com/vercel/community/assets/112093825/e1a14fa3-f9b1-4bc3-932b-6969291d8b12)
![image](https://github.com/vercel/community/assets/112093825/a5122b03-8f53-40af-9555-080dba533008)
![image](https://github.com/vercel/community/assets/112093825/89d3fa67-c560-4ca4-b16e-e14daba7ed5e)

**Reply by amyegan** [reply]
Thanks for sharing that. Looking at the code, I don't see a reason why it needs to be updated outside of build time. Can you describe the situation a bit more? I want to be sure that I understand why the page cannot be deployed as a static page so I can help you find the right solution to suit your project.

**Reply by rbhatt1999** [reply]
@amyegan  For example I add blog and push it to main it will build sitemap while building the site and at that time it takes the blogs which are existing and currently the newly added blog will not be available in the api so it is not visible in the sitemap but after redeploying it again then it is visible.
